src/sentry/utils/http.py

```diff
@@ -156,3 +158,26 @@ def is_valid_origin(origin, project=None, allowed=None):
         if parsed.path.startswith(path):
             return True
     return False
+
+
+def is_valid_ip(ip_address, project):
+    """
+    Verify that an IP address is not being blacklisted
+    for the given project.
+    """
+    blacklist = project.get_option('sentry:blacklisted-ips')
+    if not blacklist:
+        return True
+
+    ip_network = IPNetwork(ip_address)
```

Comment:
I wanted to convert this lazily, but didn't think it was that big of a deal to introduce a `SimpleLazyObject` into the mix here just for that.

Technically, it's possible that this won't ever be needed if there's never a blacklist range.

Think it's worth it? I assume most blacklists won't have a range.

I could even just do: `if '\' in addr and IPNetwork(ip_address) in IPNetwork(addr):` below and assume there won't be many ranges.